contracts/AnchoredView/AnchoredView.sol

```diff
@@ -118,58 +115,58 @@ contract AnchoredView is SymbolConfiguration {
 
             address tokenAddress = getCTokenAddress(symbol);
 
-            uint sourcePrice = priceData.getPrice(source, symbol);
+            uint reporterPrice = priceData.getPrice(reporter, symbol);
             uint anchorPrice = getAnchorInUsd(tokenAddress, usdcPrice);
 
             if (tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress || anchorPrice == 0)  {
-                emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+                emit PriceGuarded(symbol, reporterPrice, anchorPrice);
             } else {
-                uint anchorRatio = mul(sourcePrice, 100e16) / anchorPrice;
+                uint anchorRatio = mul(reporterPrice, 100e16) / anchorPrice;
                 bool withinAnchor = anchorRatio <= upperBoundAnchorRatio && anchorRatio >= lowerBoundAnchorRatio;
 
                 if (withinAnchor || !anchored) {
                     // only update and emit event if value changes
-                    if (_prices[symbol] != sourcePrice) {
-                        _prices[symbol] = sourcePrice;
-                        emit PriceUpdated(symbol, sourcePrice);
+                    if (_prices[symbol] != reporterPrice) {
+                        _prices[symbol] = reporterPrice;
+                        emit PriceUpdated(symbol, reporterPrice);
                     }
                 } else {
-                    emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+                    emit PriceGuarded(symbol, reporterPrice, anchorPrice);
                 }
             }
         }
     }
     /**
      * @notice Returns price denominated in USD, with 6 decimals
-     * @dev If price was posted by source, return it. Otherwise, return anchor price converted through source ETH price.
+     * @dev If price was posted by reporter, return it. Otherwise, return anchor price converted through reporter ETH price.
      */
     function prices(string calldata symbol) external view returns (uint) {
-        uint price = _prices[symbol];
+        CTokenMetadata memory tokenConfig = getCTokenConfig(symbol);
 
-        if (price != 0) {
-            return price;
-        } else {
+        if (tokenConfig.priceSource == PriceSource.REPORTER) return _prices[symbol];
+        if (tokenConfig.priceSource == PriceSource.FIXED_USD) return tokenConfig.fixedReporterPrice;
+        if (tokenConfig.priceSource == PriceSource.ANCHOR) {
```

Comment:
I feel like its taken for granted that anchor *must* provide prices in ETH, maybe this should be stated somewhere up front that this is the assumption?

Reply:
added a comment on the contract natspec